src/tests/integration/SparkLend.t.sol

```diff
@@ -0,0 +1,420 @@
+    function getReserveData(address asset) external view returns (ReserveData memory);
+    function supply(address asset, uint256 amount, address onBehalfOf, uint16 referralCode) external;
+    function borrow(address asset, uint256 amount, uint256 interestRateMode, uint16 referralCode, address onBehalfOf) external;
+    function repay(address asset, uint256 amount, uint256 rateMode, address onBehalfOf) external;
+    function withdraw(address asset, uint256 amount, address to) external;
+    function flashLoanSimple(
+        address receiverAddress,
+        address asset,
+        uint256 amount,
+        bytes calldata params,
+        uint16 referralCode
+    ) external;
+    function mintToTreasury(address[] calldata assets) external;
+}
+
+interface PoolAddressProviderLike {
+    function getPoolConfigurator() external view returns (PoolConfiguratorLike);
+}
+
+interface PoolConfiguratorLike {
+    function setSupplyCap(address asset, uint256 newSupplyCap) external;
+    function setReserveFlashLoaning(address asset, bool enabled) external;
+}
+
+interface DaiInterestRateStrategyLike {
+    function recompute() external;
+    function performanceBonus() external view returns (uint256);
+}
+
+interface ATokenLike {
+    function balanceOf(address) external view returns (uint256);
+    function approve(address, uint256) external returns (bool);
+    function transfer(address, uint256) external returns (bool);
+    function decimals() external view returns (uint8);
+    function RESERVE_TREASURY_ADDRESS() external view returns (address);
+    function scaledBalanceOf(address) external view returns (uint256);
+    function scaledTotalSupply() external view returns (uint256);
+    function totalSupply() external view returns (uint256);
+    function getIncentivesController() external view returns (address);
+}
+
+interface TreasuryLike {
+    function getFundsAdmin() external view returns (TreasuryAdminLike);
+}
+
+interface TreasuryAdminLike {
+    function transfer(
+        address collector,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external;
+}
+
+contract SparkLendTest is IntegrationBaseTest {
+
+    using stdJson for string;
+    using MCD for *;
+    using GodMode for *;
+    using ScriptTools for *;
+
+    PoolLike sparkPool;
+    PoolConfiguratorLike configurator;
+    DaiInterestRateStrategyLike daiInterestRateStrategy;
+    ATokenLike adai;
+    TreasuryLike treasury;
+    TreasuryAdminLike treasuryAdmin;
+    uint256 buffer;
+
+    D3MAaveTypeBufferPlan plan;
+    D3MAaveV3NoSupplyCapTypePool pool;
+
+    uint256 constant FLASHLOAN_ENABLED_MASK = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF7FFFFFFFFFFFFFFF;
+    uint256 constant SUPPLY_CAP_MASK = 0xFFFFFFFFFFFFFFFFFFFFFFFFFF000000000FFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+    uint256 constant SUPPLY_CAP_START_BIT_POSITION = 116;
+    string constant SUPPLY_CAP_EXCEEDED = '51';  // SUPPLY_CAP_EXCEEDED error code
```

Comment:
This should be removed, but maybe we can wait until be 100% sure that the supply cap is not happening.

Reply:
I've already removed the code. Can reverse the commits if need be, but it seems everyone is leaning towards no supply cap on this one.